### PR TITLE
FS-1355-assessment-progress-percentage

### PR DESCRIPTION
**Pull request by ramsharma-prog**
Just some functionality improvement

### CODE REVIEW

File: api/routes/progress_routes.py [hunk 1]
```diff
@@ -12,24 +12,15 @@ def get_progress_for_application(
...
-            scored_sub_crit_list.append(subcrit["id"]) 
-
-    # count how many of the application's scored subcrits are scored (get latest score per subcrit)
-    scored_sub_crit_list_for_application = []
-    for sub_crit in scored_sub_crit_list:
-        scored_sub_crit_for_application = get_scores_for_app_sub_crit(application_id, sub_crit)
-        scored_sub_crit_list_for_application += scored_sub_crit_for_application
 
-    number_of_scored_sub_crits_for_application = len(scored_sub_crit_list_for_application)
+    scored_sub_crit_list = [subcrit["id"] for crit in scored_criteria for subcrit in crit["sub_criteria"] if subcrit["id"] != "themes"]
```
Comment (srh-sloan) [inline api/routes/progress_routes.py:16]: Can you run the pre-commit checks? Line length is too long :)

**Review by srh-sloan — CHANGES_REQUESTED** [review]
Just formatting 👍

File: api/routes/progress_routes.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 from db.queries.scores.queries import get_scores_for_app_sub_crit
 from config.mappings.mapping_parts.scored_criteria import scored_criteria
-
+from flask import current_app
```
Comment (gio-karunakaran) [inline api/routes/progress_routes.py:5]: i think we can remove this as not being used?

**Review by gio-karunakaran — APPROVED** [review]
LGTM - just one comment

File: api/routes/progress_routes.py (hunk 2 above)
Comment (ramsharma-prog) [reply api/routes/progress_routes.py:5]: Yea will do

**Review by srh-sloan — APPROVED** [review]
👍

File: api/routes/assessment_routes.py [hunk 3]
```diff
@@ -2,6 +2,8 @@
 from typing import Dict
 from typing import List
 
+from api.routes.progress_routes import get_progress_for_application
```
Comment (johnoneill46) [inline api/routes/assessment_routes.py:5]: Is this a redundant import?

Comment (gio-karunakaran) [reply api/routes/assessment_routes.py]: removed 👍🏾

File: db/queries/progress/queries.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    
+    progress_metadatas = [
+        metadata_serialiser.dump(
+            {
+                "application_id": app_id, 
+                "scored_sub_criterias": score_count
+            }) for app_id, score_count in results
+        ]
+
+    return progress_metadatas
```
Comment (ramsharma-prog) [inline db/queries/progress/queries.py:37]: Looks good ❤️

**Pull request merged**